File: Perhotelan_C#/Perhotelan/f_edit_tamu.cs
using System;
using System.Data;
using System.Data.Odbc;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace Perhotelan
{
    public partial class f_edit_tamu
    {
        public f_edit_tamu()
        {
            InitializeComponent();
            _Panel1.Name = "Panel1";
            _L_header.Name = "L_header";
            _Label1.Name = "Label1";
            _T_nama.Name = "T_nama";
            _Label2.Name = "Label2";
            _Label3.Name = "Label3";
            _T_alamat.Name = "T_alamat";
            _Label4.Name = "Label4";
            _T_telp.Name = "T_telp";
            _Label5.Name = "Label5";
            _C_jk.Name = "C_jk";
            _Label6.Name = "Label6";
            _Label7.Name = "Label7";
            _Label8.Name = "Label8";
            _Label9.Name = "Label9";
            _Label10.Name = "Label10";
            _B_simpan.Name = "B_simpan";
            _B_reset.Name = "B_reset";
            _L_id.Name = "L_id";
        }

        private OdbcConnection conn;
        private OdbcCommand cmd;
        private OdbcDataReader dr;
        private OdbcDataAdapter da;
        private DataTable dt;
        private DataSet ds;

        public void koneksi()
        {
            try
            {
                conn = new OdbcConnection();
                conn.ConnectionString = "dsn=db_hotel";
                conn.Open();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Koneksi Ke Database Gagal", Constants.vbInformation, "Pemberitahuan");
            }
        }

        public void reset()
        {
            T_nama.Clear();
            C_jk.SelectedIndex = 0;
            T_alamat.Clear();
            T_telp.Clear();
        }

        public void simpanUpdate()
        {
            koneksi();
            try
            {
                string sql = "update tb_tamu set" + " nama_tamu='" + T_nama.Text + "', " + " jenis_kelamin='" + C_jk.Text + "', " + " alamat='" + T_alamat.Text + "', " + " no_telp='" + T_telp.Text + "'" + " where id='" + L_id.Text + "'";
                cmd = new OdbcCommand(sql, conn);
                cmd.ExecuteNonQuery();
                reset();
                My.MyProject.Forms.f_tamu.tampil_tamu();
                Close();
                Interaction.MsgBox("Berhasil Edit Data", Constants.vbInformation, "Pemberitahuan");
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Gagal Edit Data", Constants.vbInformation, "Pemberitahuan");
            }
        }

        public void ketemu()
        {
            koneksi();
            try
            {
                cmd = new OdbcCommand("select * from tb_tamu where id='" + L_id.Text + "'", conn);
                dr = cmd.ExecuteReader();
                dr.Read();
                if (dr.HasRows)
                {
                    T_nama.Text = Conversions.ToString(dr["nama_tamu"]);
                    C_jk.Text = Conversions.ToString(dr["jenis_kelamin"]);
                    T_alamat.Text = Conversions.ToString(dr["alamat"]);
                    T_telp.Text = Conversions.ToString(dr["no_telp"]);
                }
                else
                {
                    Close();
                    Interaction.MsgBox("Data Tidak Ditemukan", Constants.vbInformation, "Pemberitahuan");
                }
            }
            catch (Exception ex)
            {
            }
        }

        private void f_edit_tamu_Load(object sender, EventArgs e)
        {
            C_jk.SelectedIndex = 0;
            ketemu();
        }

        private void B_reset_Click(object sender, EventArgs e)
        {
            reset();
        }

        private void B_simpan_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(T_nama.Text))
            {
                Interaction.MsgBox("Kolom Nama Masih Kosong!", Constants.vbInformation, "Pemberitahuan");
                T_nama.Focus();
            }
            else if (!(C_jk.SelectedIndex == 1 | C_jk.SelectedIndex == 2))
            {
                Interaction.MsgBox("Kolom Jenis Kelamin Masih Kosong!", Constants.vbInformation, "Pemberitahuan");
                C_jk.Focus();
            }
            else if (string.IsNullOrEmpty(T_alamat.Text))
            {
                Interaction.MsgBox("Kolom Alamat Masih Kosong!", Constants.vbInformation, "Pemberitahuan");
                T_alamat.Focus();
            }
            else if (string.IsNullOrEmpty(T_telp.Text))
            {
                Interaction.MsgBox("Kolom No. Telp Masih Kosong!", Constants.vbInformation, "Pemberitahuan");
                T_telp.Focus();
            }
            else
            {
                koneksi();
                try
                {
                    cmd = new OdbcCommand("select * from tb_tamu where id='" + L_id.Text + "'", conn);
                    dr = cmd.ExecuteReader();
                    dr.Read();
                    if (dr.HasRows)
                    {
                        simpanUpdate();
                    }
                    else
                    {
                        Interaction.MsgBox("Data Tidak Ditemukan", Constants.vbInformation, "Pemberitahuan");
                        Close();
                    }
                }
                catch (Exception ex)
                {
                }
            }
        }
    }
}

File: Perhotelan_VB/Perhotelan/f_edit_tamu.vb
Imports System.Data.Odbc

Public Class f_edit_tamu

    Dim conn As OdbcConnection
    Dim cmd As OdbcCommand
    Dim dr As OdbcDataReader
    Dim da As OdbcDataAdapter
    Dim dt As DataTable
    Dim ds As DataSet

    Sub koneksi()
        Try
            conn = New OdbcConnection
            conn.ConnectionString = "dsn=db_hotel"
            conn.Open()
        Catch ex As Exception
            MsgBox("Koneksi Ke Database Gagal", vbInformation, "Pemberitahuan")
        End Try
    End Sub

    Sub reset()
        T_nama.Clear()
        C_jk.SelectedIndex = 0
        T_alamat.Clear()
        T_telp.Clear()
    End Sub

    Sub simpanUpdate()
        koneksi()
        Try
            Dim sql As String = "update tb_tamu set" +
                                " nama_tamu='" + T_nama.Text + "', " +
                                " jenis_kelamin='" + C_jk.Text + "', " +
                                " alamat='" + T_alamat.Text + "', " +
                                " no_telp='" + T_telp.Text + "'" +
                                " where id='" + L_id.Text + "'"
            cmd = New OdbcCommand(sql, conn)
            cmd.ExecuteNonQuery()
            reset()
            f_tamu.tampil_tamu()
            Me.Close()
            MsgBox("Berhasil Edit Data", vbInformation, "Pemberitahuan")
        Catch ex As Exception
            MsgBox("Gagal Edit Data", vbInformation, "Pemberitahuan")
        End Try
    End Sub

    Sub ketemu()
        koneksi()
        Try
            cmd = New OdbcCommand("select * from tb_tamu where id='" + L_id.Text + "'", conn)
            dr = cmd.ExecuteReader
            dr.Read()
            If dr.HasRows Then
                T_nama.Text = dr.Item("nama_tamu")
                C_jk.Text = dr.Item("jenis_kelamin")
                T_alamat.Text = dr.Item("alamat")
                T_telp.Text = dr.Item("no_telp")
            Else
                Me.Close()
                MsgBox("Data Tidak Ditemukan", vbInformation, "Pemberitahuan")
            End If
        Catch ex As Exception

        End Try
    End Sub

    Private Sub f_edit_tamu_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        C_jk.SelectedIndex = 0
        ketemu()
    End Sub

    Private Sub B_reset_Click(sender As Object, e As EventArgs) Handles B_reset.Click
        reset()
    End Sub

    Private Sub B_simpan_Click(sender As Object, e As EventArgs) Handles B_simpan.Click
        If T_nama.Text = "" Then
            MsgBox("Kolom Nama Masih Kosong!", vbInformation, "Pemberitahuan")
            T_nama.Focus()
        ElseIf Not (C_jk.SelectedIndex = 1 Or C_jk.SelectedIndex = 2) Then
            MsgBox("Kolom Jenis Kelamin Masih Kosong!", vbInformation, "Pemberitahuan")
            C_jk.Focus()
        ElseIf T_alamat.Text = "" Then
            MsgBox("Kolom Alamat Masih Kosong!", vbInformation, "Pemberitahuan")
            T_alamat.Focus()
        ElseIf T_telp.Text = "" Then
            MsgBox("Kolom No. Telp Masih Kosong!", vbInformation, "Pemberitahuan")
            T_telp.Focus()
        Else
            koneksi()
            Try
                cmd = New OdbcCommand("select * from tb_tamu where id='" + L_id.Text + "'", conn)
                dr = cmd.ExecuteReader
                dr.Read()
                If dr.HasRows Then
                    simpanUpdate()
                Else
                    MsgBox("Data Tidak Ditemukan", vbInformation, "Pemberitahuan")
                    Me.Close()
                End If
            Catch ex As Exception

            End Try
        End If
    End Sub

End Class